### PR TITLE
(PC-9969) App Native - Amélioration tech - fix upgrade react-native

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-9969

## Done 

- Commented `node_modules/react-native/scripts/find-node.sh` with patch-package.
- Upgrade flipper & co as recommended in https://github.com/aiba/react-native-m1/blob/main/README.md and https://github.com/facebook/react-native/issues/31480
- Added `react_native_post_install(installer)` in `Podfile`: https://stackoverflow.com/a/67336115/2127277

## Alternative. to keep the upgrade

- keep react-native 0.64.2 and disable hermes
- keep react-native 0.64.2, keep hermes, disable Flipper in Podfile

## Unnecessary fixed

- we did not add the Apple M1 Podfile fix, we'll do if necessary
- seems like flipper does not need the extra Podfile fix replacement anymore, at least in my case

## Potential issue

- If CI/Fastlane relies on nvm and nvm is not initialized in the shell process, it may break, but it is less likely to happen, wait and see.

## Checklist

I have:

- [ ] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [ ] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Written **documentation on Notion** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: ios/Podfile [hunk 1]
```diff
@@ -16,7 +16,7 @@ target 'PassCulture' do
   #
   # Note that if you have use_frameworks! enabled, Flipper will not work and
   # you should disable these next few lines.
-  use_flipper!({ 'Flipper' => '0.92.0' , 'Flipper-Folly' => '~> 2.5'})
+  use_flipper!({ 'Flipper' => '0.95.0', 'Flipper-Folly' => '2.6.7', 'Flipper-DoubleConversion' => '3.1.7' })
```
Comment (kopax-polyconseil) [inline ios/Podfile:19]: le combo viens de https://github.com/aiba/react-native-m1/blob/main/README.md et https://github.com/facebook/react-native/issues/31480 , downgrade ne fonctionne pas.

File: ios/Podfile [hunk 2]
```diff
@@ -34,6 +34,8 @@ target 'PassCulture' do
 
   post_install do |installer|
     flipper_post_install(installer)
+    react_native_post_install(installer)
```
Comment (kopax-polyconseil) [inline ios/Podfile:37]: Sans cette ligne, on rencontre une erreur de build.

**Review by BenStirrup — APPROVED** [review]

**Review by marionnousvalentin — APPROVED** [review]
Top !

**Comment by airtonix** [comment]
you guys should remove nvm from your chain and just use asdf, asdf-nodejs.

 react-native will behave as if you have a global install of node  this way.

**Pull request merged**